### ISSUE TITLE
.RecreateStoreOnModelMismatch does not work

### ISSUE THREAD

**Issue by TechMaster** [JohnEstropia/CoreStore#96]
I found that  .RecreateStoreOnModelMismatch does not work. When I configured storage, then added new entity to CoreData model, then I found error.

I also tried with demo source code in github and encounter same problem.
**"The model used to open the store is incompatible with the one used to create the store"**

```
func configureStorage() {
        _ = dataStack.addStorage(
            SQLiteStore(
                fileName: "HumanResource.sql",
                configuration: "Default",
                localStorageOptions: .RecreateStoreOnModelMismatch)
            ,
            completion: { (result) in
                switch result {
                case .Success(let storage):
                    print("Successfully added sqlite store: \(storage)")
                case .Failure(let error):
                    print("Failed adding sqlite store with error: \(error)")
                }
        })
        
        CoreStore.defaultStack = dataStack
    }
```

**Comment by JohnEstropia** [comment]
Can you reproduce this on iOS 8? I feel this is due to iOS 9's persistent store model caching.

**Comment by TechMaster** [comment]
I am download iOS 8 simulator now. But CoreStore crashes in iOS 9 when developer change model : adding new entity, modify new attribute.

Is there any option to turn off  iOS 9's persistent store model caching?

**Comment by JohnEstropia** [comment]
Not that I've seen yet. I'll see if there's a way to work around it. Thanks for the report!

**Comment by TechMaster** [comment]
Hi John,

I simulate on iOS 8.4 encounter sample problem. I made a simple project so you can easily reproduce bug at your side in 5'.

Please pull code at https://github.com/TechMaster/SimpleCoreStore

⚠️ [CoreStore: Warning] From.swift:204 applyToFetchRequest(_:context:applyAffectedStores:) ↪︎ Attempted to perform a fetch but could not find any persistent store for the entity

❗ [CoreStore: Fatal Error] BaseDataTransaction.swift:82 create ↪︎ Attempted to create an entity of type 'Person', but a destination persistent store containing the entity type could not be found.

**Comment by JohnEstropia** [comment]
Thank you for your report. I'm afraid I can't reproduce this If I don't have your old xcdatamodeld. You can check that with your app too, if you delete and reinstall your app you won't see this error again (until the next time you change the model). But the cause of this problem is clear so I'll try look for a fix.

**Comment by TechMaster** [comment]
To reproduce error, you just create a new entity in model, you will see error immediately.

For the moment, I work around by deleting the app when model is changed. I think that .RecreateStoreOnModelMismatch does not delete model file.

**Comment by JohnEstropia** [comment]
@TechMaster Please try out the fix I pushed to the `develop` branch. If it fixes the problem for you I'll push it to a Cocoapods update

**Comment by TechMaster** [comment]
Hi John,

It works perfectly !
I pull your latest source code from develop branch, copy  all files, folders in Sources folder then paste to folder Pods\CoreStore\Sources. I compiled and made change to model. The app did not crash any more, it delete old model and copy new compile model.

You can push new code to Cocoapods now. After you push, please let me now, i will test again.

Thanks

**Comment by JohnEstropia** [comment]
Fixed in version 2.0.6. Thanks again!
